saleor/checkout/migrations/0056_move_checkout_metadata_to_separate_model.py

```diff
@@ -0,0 +1,64 @@
+from django.db import migrations
+from django.db.models import Q
+
+
+BATCH_SIZE = 10000
+
+
+def queryset_in_batches(queryset):
+    """Slice a queryset into batches.
+
+    Input queryset should be sorted by pk.
+    """
+    start_pk = 0
+
+    while True:
+        qs = queryset.filter(pk__gt=start_pk)[:BATCH_SIZE]
+        pks = list(qs.values_list("pk", flat=True))
+
+        if not pks:
+            break
+
+        yield pks
+
+        start_pk = pks[-1]
+
+
+def clear_checkout_metadata(checkout):
+    checkout.metadata = None
+    checkout.private_metadata = None
+
+
+def move_all_checkout_metadata(apps, schema_editor):
+    Checkout = apps.get_model("checkout", "Checkout")
+    CheckoutMetadata = apps.get_model("checkout", "CheckoutMetadata")
+    checkouts_with_meta = (
+        Checkout.objects.filter(
+            Q(metadata__isnull=False) | Q(private_metadata__isnull=False)
+        )
+        .order_by("pk")
+        .distinct("token")
```

Comment:
The checkout token is a PK for checkout, do we need the distinct?

Reply:
also, we should use `only` method on a QS to limit the data fetched and mem usage (since we don't care at all about most of the fields)
`.only("pk", "metadata", "private_metadata")`

Reply:
distinct is used so we don't get duplicates for those Checkout object that has both metadata and private_metadata. 
Great catch with `only`, I'll fix that

Reply:
`.distinct` means there is a missing `EXISTS` subquery, and we're merging an M2M table instead.

Reply:
@patrys fields I'm filtering with here (`metadata` and `private_metadata`) are both fields of Checkout model. I'm not sure about which M2M table merging you're mentioning. This query (with added `only` mentioned by Tomasz) will look like in the screenshot: 
![image](https://user-images.githubusercontent.com/34745469/202143044-ce34ccc3-ca39-4e73-ad7d-a0382c3da56c.png)

Reply:
How can you get duplicate results when you're querying a single table?